examples/mxnet/gcn/gcn_spmv.py

```diff
@@ -0,0 +1,187 @@
+        if self.dropout:
+            emb_inp[-1] = mx.nd.Dropout(emb_inp[-1], p=self.dropout)
+
+        self.g.ndata['h'] = mx.nd.concat(*emb_inp, dim=1)
+        for layer in self.conv_layers:
+            if self.normalization == 'sym':
+                self.g.ndata['h'] = self.g.ndata['h'] / self.g.ndata['out_degree'].sqrt()
+            self.g.update_all(fn.copy_src(src='h', out='m'),
+                              fn.sum(msg='m', out='accum'),
+                              layer)
+
+        emb_out = self.g.ndata.pop('h')
+        return self.out_layer(emb_out)
+
+
+def main(args):
+    # load and preprocess dataset
+    data = load_data(args)
+
+    if args.self_loop:
+        data.graph.add_edges_from([(i,i) for i in range(len(data.graph))])
+
+    features = mx.nd.array(data.features)
+    labels = mx.nd.array(data.labels)
+    mask = mx.nd.array(data.train_mask)
+    in_degree  = mx.nd.array([data.graph.in_degree(i)
+                              for i in range(len(data.graph))])
+    out_degree = mx.nd.array([data.graph.out_degree(i)
+                              for i in range(len(data.graph))])
+
+    in_feats = features.shape[1]
+    n_classes = data.num_labels
+    n_edges = data.graph.number_of_edges()
+
+    if args.gpu <= 0:
+        cuda = False
+        ctx = mx.cpu(0)
+    else:
+        cuda = True
+        features = features.as_in_context(mx.gpu(0))
+        labels = labels.as_in_context(mx.gpu(0))
+        mask = mask.as_in_context(mx.gpu(0))
+        in_degree = in_degree.as_in_context(mx.gpu(0))
+        out_degree = out_degree.as_in_context(mx.gpu(0))
+        ctx = mx.gpu(0)
+
+    # create GCN model
+    g = DGLGraph(data.graph)
+    g.ndata['in_degree']  = in_degree.reshape((-1,1))
+    g.ndata['out_degree'] = out_degree.reshape((-1,1))
+
+    model = GCN(g,
+                in_feats,
+                args.n_hidden,
+                n_classes,
+                args.n_layers,
+                'relu',
+                args.dropout,
+                args.normalization,
+                )
+    model.initialize(ctx=ctx)
+    loss_fcn = gluon.loss.SoftmaxCELoss()
+
+    # use optimizer
+    trainer = gluon.Trainer(model.collect_params(), 'adam', {'learning_rate': args.lr})
+
+    # initialize graph
+    dur = []
+    for epoch in range(args.n_epochs):
+        if epoch >= 3:
+            t0 = time.time()
+        # forward
+        with mx.autograd.record():
+            pred = model(features)
+            loss = loss_fcn(pred, labels, mask)
+
+        #optimizer.zero_grad()
+        loss.backward()
+        trainer.step(features.shape[0])
+
+        if epoch >= 3:
+            dur.append(time.time() - t0)
+            print("Epoch {:05d} | Loss {:.4f} | Time(s) {:.4f} | ETputs(KTEPS) {:.2f}".format(
+                epoch, loss.asnumpy()[0], np.mean(dur), n_edges / np.mean(dur) / 1000))
+
+    # test set accuracy
+    pred = model(features)
+    accuracy = (pred*100).softmax().pick(labels).mean()
```

Comment:
this accuracy includes the training data. could you please avoid using the training data for test?